packages/app/components/token-picker/TokenPicker.tsx

```diff
@@ -1,69 +1,105 @@
+import { Token } from "@/models/token/types";
 import {
   BodyText,
+  Icon,
   Modal,
   ModalBaseProps,
   ModalContent,
   ModalHeaderTitle,
+  RadioButton
 } from "@/ui";
+import { TokenIcon } from "../TokenIcon";
+import { TOKEN_PICKER_COMMON_TOKENS } from "./constants";
+
+interface TokenListRowProps {
+  token: Token;
+  closeAction: () => void;
+}
+
+interface TokenListProps {
+  closeAction: () => void;
+  tokenList: Token[];
+}
 
 export const TokenPicker = ({ isOpen, closeAction }: ModalBaseProps) => {
+  const { tokenList } = useTokenListContext();
+
   return (
-    <div>
-      <Modal isOpen={isOpen} closeAction={closeAction}>
-        <ModalHeaderTitle title="Select a token" closeAction={closeAction} />
-        <ModalContent>
-          <div className="space-y-4">
-            <input
-              type="search"
-              placeholder="search token name or paste address"
-              className="w-full p-2 bg-surface-75 rounded-xl"
-            />
-            <div className="py-3 space-y-2">
-              <p className="text-xs font-semibold text-em-low">Common tokens</p>
-              <div className="flex items-center space-x-3">
-                <div className="p-2 font-semibold uppercase border text-em-med w-fit rounded-2xl">
-                  Eth
-                </div>
-                <div className="p-2 font-semibold uppercase border text-em-med w-fit rounded-2xl">
-                  Pepe
-                </div>
-                <div className="p-2 font-semibold uppercase border text-em-med w-fit rounded-2xl">
-                  usdc
-                </div>
-              </div>
-            </div>
-            <div className="relative">
-              <div className="-mx-4 overflow-y-auto border-t h-72 border-surface-50">
-                {[
-                  "1Inch",
-                  "Aave",
-                  "DAI",
-                  "USDC",
-                  "Matic",
-                  "Uni",
-                  "pepe",
-                  "Swapr",
-                  "ARB",
-                ].map((token) => (
-                  <div
-                    className="flex justify-between w-full px-4 py-2 border-b cursor-pointer border-surface-50 hover:bg-surface-50"
-                    key={token}
-                    onClick={closeAction}
-                  >
-                    <div className="flex items-center space-x-3">
-                      <div className="w-8 h-8 rounded-full bg-primary-100" />
-                      <BodyText>{token}</BodyText>
-                    </div>
-                    <BodyText>0</BodyText>
-                  </div>
-                ))}
-              </div>
-            </div>
-          </div>
-        </ModalContent>
-      </Modal>
-    </div>
+    <Modal closeAction={closeAction} isOpen={isOpen}>
+      <ModalHeaderTitle closeAction={closeAction} title="Select a token" />
+      <ModalContent className="mt-3 pt-3 px-0">
```

Comment:
This style seem odd: `mt-3` and `pt-3`

I think it has been added related to changing the styles on `ModalHeader`.
I didn't understand why we change those and why we need this.

Reply:
Oh, just to replicate a bit the look&feel, the last change requested by @berteotti was to use `divide-y` and divide colors to emulate horizontal dividers.

As `divide-y` styles the top border for each child element of a container, basically is the top border for the element box. I used top margin (to distance the box top border)  from the Modal Header, and the padding to distance the SearchBar from the box top border

Reply:
As discussed offline between @Diogomartf and @berteotti , we decided to revert these changes on the modal dividers, keep the `divide-` classes for the `TokenPickerList` for now, and create a follow-up task to assess how to approach a normalization on how to handle dividers for components, starting for modals.

Related: https://linear.app/swaprdev/issue/STK-90/modal-component-dividers

Reply:
As discussed offline between @Diogomartf and @berteotti , we decided to revert these changes on the modal dividers, keep the `divide-` classes for the `TokenPickerList` for now, and create a follow-up task to assess how to approach a normalization on how to handle dividers for components, starting for modals.

Related: https://linear.app/swaprdev/issue/STK-90/modal-component-dividers